<a href="https://colab.research.google.com/github/Mariett-7/programacion1/blob/main/quiz%2026%20marzo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

QUIZ 26_MARZO

ANGIE MARIETH ORTEGA JAIMES

In [66]:
%%writefile pedirnotas.py
def pedir_nombre():
  return input("ingrese el nombre del estudiante: ")

def pedir_notas():
    n1 = float(input("nota 1: "))
    n2 = float(input("nota 2: "))
    n3 = float(input("nota 3: "))
    n4 = float(input("nota 4: "))
    return n1, n2, n3, n4

Overwriting pedirnotas.py


In [61]:
%%writefile promedio.py
import statistics
def calcular_promedio(n1, n2, n3, n4):
  return (n1+n2+n3+n4)/4

def mediana (n1, n2, n3, n4):
  return statistics.median([n1, n2, n3, n4])

def moda (n1, n2,n3,n4):
   return statistics.mode([n1, n2, n3, n4])

Overwriting promedio.py


In [67]:
import importlib
import pedirnotas
importlib.reload(pedirnotas)
import promedio
importlib.reload(promedio)
pedirnotas.pedir_nombre()

ingrese el nombre del estudiante: angie


'angie'

In [68]:
notas = pedirnotas.pedir_notas()


nota 1: 2
nota 2: 5
nota 3: 4
nota 4: 3


In [74]:
import promedio
import importlib

importlib.reload(promedio)

nombre = pedirnotas.pedir_nombre()

prom = promedio.calcular_promedio(n1, n2, n3, n4)
mediana = promedio.mediana(n1, n2, n3, n4)
moda = promedio.moda(n1, n2, n3, n4)

print("estudiante:", nombre)
print("promedio:", prom)
print("mediana:", mediana)
print("moda:", moda)

if prom >= 3:
  print("aprobado")
else:
  print("reprobado")

ingrese el nombre del estudiante: angie
estudiante: angie
promedio: 3.5
mediana: 3.5
moda: 2.0
aprobado


en promedio el estudiante obtuvo 3.5 y se encuentra en el 50%  de la calificacion, pues esta tambien es de 3.5. la calificación más repetida fue en 2.0. Así el desempeño del estudiante es bueno en comparación a la moda de estás calificaciones y se encuentra en estado aprobado.